framework/src/transfers/MultiAppMeshFunctionTransfer.C

```diff
@@ -228,24 +258,24 @@ MultiAppMeshFunctionTransfer::execute()
   // Send points to other processors.
   std::vector<std::vector<Real>> incoming_evals(n_processors());
   std::vector<std::vector<unsigned int>> incoming_app_ids(n_processors());
-  std::vector<Parallel::Request> send_points(n_processors());
-  for (processor_id_type i_proc = 0; i_proc < n_processors(); i_proc++)
+  _send_points.resize(n_processors());
+  for (processor_id_type i_proc = 0; i_proc < n_processors(); ++i_proc)
   {
     if (i_proc == processor_id())
       continue;
-    _communicator.send(i_proc, outgoing_points[i_proc], send_points[i_proc]);
+    _communicator.send(i_proc, outgoing_points[i_proc], _send_points[i_proc]);
   }
 
-  // Recieve points from other processors, evaluate mesh frunctions at those
+  // Receive points from other processors, evaluate mesh functions at those
   // points, and send the values back.
-  std::vector<Parallel::Request> send_evals(n_processors());
-  std::vector<Parallel::Request> send_ids(n_processors());
+  _send_evals.resize(n_processors());
+  _send_ids.resize(n_processors());
```

Comment:
It's unclear to me if you've introduced a bug here. I can see why you elevated these variable to instance variables but in doing so the state of these can stick around between invocations of this object. We may need to cleanup or reinitialize these values before each iteration. 

@roystgnr - maybe you can quickly answer this. Do these Parallel::Request objects need to be reinitialized each time they are used? The old code would create them in some default state, the new code would just reuse those locations and I'm unsure if that's equivalent.

Reply:
I see; for safety, I can add:

```
_send_evals.clear();
_send_ids.clear();
```

Reply:
The default state isn't actually super important, all that's important is that the requests aren't leaked.  See https://stackoverflow.com/questions/22410827/mpi-reuse-mpi-request

With Parallel::Request, if you wait() on a request it won't be leaked, but we *don't* currently have any automatic protection against leaks.

Reply:
I suppose I could automatically free a request with MPI_Cancel() and MPI_Request_free() in the Parallel::Request::cleanup(), I suppose...  but I'm not sure that would be the safest thing to do... I'm actually tempted to go the other way, and test during cleanup() to make sure that a wait() has already been done, after providing a cancel() method for users who really do change their minds.

Reply:
OK - so the takeaway is that this interface is a little dangerous and we can definitely shoot ourselves in the foot, but hey, this is C++ and we are used to that. Doesn't sound like we need the clear() (you wouldn't want that anyway) in case it causes allocation, deallocation slowdowns. If we did need to reinitialize (which it sounds like we don't) we'd either call a method on the object to do that, or use vector::assign() to clean it up before the next round.

Reply:
ok, so no change is requested here, right?

Reply:
This does look safe to me, with no change required, although I wouldn't swear to it, since (as my other comment indicated) I'm having a little trouble following the expected flow of execution.